pytest_motor/mongod_binary.py

```diff
@@ -0,0 +1,151 @@
+                    while True:
+                        # read by 100 bytes
+                        chunk = await resp.content.read(100)
+                        if not chunk:
+                            break
+                        binary_file.write(chunk)
+                    binary_file.flush()
+                    binary_file.seek(0)
+                    self.__unpack(binary_file)
+                    return self.path
+
+    def __unpack(self, binary_file: IO[bytes]) -> None:
+        """Unpack mongod binary from zip or tar."""
+        if self.url.endswith('.tgz'):
+            with tarfile.open(fileobj=binary_file) as archive_tar:
+                file_in_archive = archive_tar.extractfile(
+                    f'mongodb-{self.current_platform}-{self.MONGO_VERSION}/bin/mongod')
+                if file_in_archive is None:  # pragma: no cover
+                    raise FileNotFoundError
+                file_outside = self.path.open(mode="wb")
+                with file_in_archive, file_outside:
+                    shutil.copyfileobj(file_in_archive, file_outside)
+                self.path.chmod(0o700)
+            assert self.path.exists(), "Unsuccessful mongod binary extraction"
+            return
+        if self.url.endswith('.zip') and self.current_platform == 'windows-x86_64':
+            with ZipFile(file=binary_file, mode='r') as archive_zip:
+                # pylint: disable=consider-using-with
+                file_in_archive = archive_zip.open(
+                    f'mongodb-win32-x86_64-windows-{self.MONGO_VERSION}/bin/mongod.exe')
+                if file_in_archive is None:  # pragma: no cover
+                    raise FileNotFoundError
+                file_outside = self.path.open(mode="wb")
+                with file_in_archive, file_outside:
+                    shutil.copyfileobj(file_in_archive, file_outside)
+                self.path.chmod(0o700)
+            assert self.path.exists(), "Unsuccessful mongod binary extraction"
+            return
+        raise Exception("Unsupported archive format.")  # pragma: no cover
+
+    @staticmethod
+    def __select_ubuntu_version() -> str:
+        if bool(distro.major_version()) and int(distro.major_version()) < 16:
+            raise OSError("Your Ubuntu version is too old. Upgrade at least to 16.")
+
+        if distro.version() == '16.04':
+            MongodBinary.warn_untested_os()
+            return 'ubuntu1604'
+        if distro.version() == '18.04':
+            return 'ubuntu1804'
+        if distro.version() == '20.04':
+            return 'ubuntu2004'
+
+        warnings.warn("Can't detect your Ubuntu version. Fallback to 18.04.")
+        return 'ubuntu1804'
+
+    @staticmethod
+    def __select_debian_version() -> str:
+        if bool(distro.major_version()) and int(distro.major_version()) < 9:
+            raise OSError(
+                "Your Debian version is too old. Upgrade at least to 9.")  # pragma: no cover
```

Comment:
I don't know why, but `pytest` reports this as uncovered.

Reply:
*was reporting, that's why I added `pragma: no cover`

Reply:
I think it is probably a bug with the line continuation